$OpenBSD: patch-Makefile,v 1.2 2021/08/28 09:07:11 cwen Exp $

The `-lm' linker flag is needed for architectures not inlining sqrt(3), like
powerpc.

Index: Makefile
--- Makefile.orig
+++ Makefile
@@ -1,26 +1,20 @@
 # Makefile for test program for game_libs - lunar lander
-CFLAGS=-Wall `sdl-config --libs --cflags` 
-CC=gcc
+CFLAGS+=`sdl-config --cflags` 
+#CC=gcc
 
-LIBS=SDL_image
+LIBS=`sdl-config --libs` -lSDL_image -lSDL_mixer -lm
 
 C_FILES=moon_lander.c game_lib.c DT_drawtext.c
 OBJ_FILES=moon_lander.o game_lib.o DT_drawtext.o 
-OUT_FILE=moon-lander.bin
+OUT_FILE=moonlander
 
 all: game_lib
 
 game_lib: $(OBJ_FILES)
-	$(CC) $(CFLAGS) -o $(OUT_FILE) $(OBJ_FILES) -l$(LIBS) -lSDL_mixer
+	$(CC) $(CFLAGS) -o $(OUT_FILE) $(OBJ_FILES) $(LIBS)
 
-moon_lander.o: moon_lander.c
-	$(CC) $(CFLAGS) -c -o $@ $^
-
-game_lib.o: game_lib.c
-	$(CC) $(CFLAGS) -c -o $@ $^
-
-DT_drawtext.o: DT_drawtext.c
-	$(CC) $(CFLAGS) -c -o $@ $^
+.c.o:
+	$(CC) $(CFLAGS) -c $< -o $@
 
 clean: 
 	rm -f *.o core  
